src/components/DonateMoney.vue

```diff
@@ -25,22 +83,75 @@
 <script lang="js">
 import axios from 'axios'
 export default {
-    name: 'DonateMoney',
-    props: [],
-    data () {
-      return {
-        donation: 1.50
-      }
+  name: 'DonateMoney',
+  props: [],
+  data () {
+    return {
+      donationAmount: 2,
+      customAmountSelected: false,
+      customDonationAmount: undefined,
+      message: ''
+    }
+  },
+  computed: {
+  },
+  mounted () {
+  },
+  methods: {
+    unsetCustomAmountSelection() {
+      this.customAmountSelected = false;
+      this.message = "";
     },
-    computed: {
+    handleCustomAmountSelection() {
+      this.customAmountSelected = !this.customAmountSelected;
+      this.message = "";
     },
-    mounted () {
+    validateDonationAmount(value = this.customDonationAmount) {
+      let isValid = true;
+      value = parseFloat(value); // outputs: number
+      value = value.toFixed(2); // outputs: string
+      value = parseFloat(value); // outputs: number
+      // separating parameter assignment and parseFloat operation for consistent outcome on change
+
+      // if an existing donation amount button was chosen
+      if (!this.customAmountSelected) {
+        value = this.donationAmount;
+        return { value, isValid } // i.e. { 2 || 20 || 50, true }
+      }
+
+      if (value > 1.49 && value < 10000.01) {
+        this.message = "";
+        return { value, isValid }; // { 1.50, true }
+      }
+
+      if (isNaN(value)) { // undefined || NaN
+        this.message = "Please select or enter a valid amount.";
+      }
+      if (value < 1.5) {
+        this.message = "Please enter a donation amount higher than $1.50.";
+      }
+      if (value > 10000) {
+        this.message = "Amplify currently only accept donation amounts less than $10,000."
+      }
+
+      isValid = false;
+      return { value, isValid }; // i.e. { undefined, false }
+    },
+    submit() {
+      const { value, isValid } = this.validateDonationAmount();
+      // console.log(`the value: ${value} is ${isValid ? 'valid' : 'NOT valid'}`);
+
+      if (isValid) {
+        this.createCheckoutSession(value);
+        this.message = ""; // only refreshes message after input value becomes valid
+      } else {
+        this.message = "The donation amount is not valid."; // edge case?
+        return;
+      }
     },
-    methods: {
-      submit () {
-        axios.post('/api/checkout/create-checkout-session', {donationAmount: this.donation})
+    createCheckoutSession (donationAmount) {
```

Comment:
Very nice. The whole front-end appears to be working correctly. Great job handling the cases. There appears to be some work left in the `create-checkout-session` call -- namely, a check for valid donation amount that must be changed.

Reply:
I was going to work on that in a different PR. Would you like me to make that change here?

Reply:
Made some changes in `checkout.js` and refactored for readability. Hope you like it!